### PR TITLE
Add new test case: Download from a helm repo with a secret

### PR DESCRIPTION
**Pull request by pezhang**
New test case: Download from a helm repo with a reference to a secret that has “user” and “password” defined for basic authentication GET request

### CODE REVIEW

**Comment by pezhang** [comment]
Basic authentication requires username and password to get access to http or https. I tried with below public websites, but they don't support basic authentication. So I created a Apache server enabling the basic authentication, then put the chart file into it. Below is the steps. 

Public website not support basic auth:
1.Bugzilla: didn't found related function
2.Github: didn't support basic auth, refer to https://docs.github.com/en/free-pro-team@latest/rest/overview/other-authentication-methods 
3.Google drive : didn't find related function

Steps to create Apache server to enable basic auth:
1. Install httpd
# sudo yum install -y httpd
# sudo systemctl start httpd

2. Put the chart file to below 
# sudo mkdir /var/www/html/helmreleaseAPItest/

3. Create user&password file
# htpasswd -cm /var/www/html/helmreleaseAPItest/.htpasswd user
New password: (redhat)
Re-type new password:(redhat)
Adding password for user helmreleaseapi-user

# ls /var/www/html/helmreleaseAPItest/.htpasswd
/var/www/html/helmreleaseAPItest/.htpasswd

# cat /var/www/html/helmreleaseAPItest/.htpasswd
user:$apr1$47XP0iTk$DvQJHkXDE/9I8hE/1.SIX1

4. Add/Change access config files

# ll /var/www/html/helmreleaseAPItest/.htaccess
-rw-r--r--. 1 root root 114 Dec 21 11:19 /var/www/html/helmreleaseAPItest/.htaccess

# cat /var/www/html/helmreleaseAPItest/.htaccess 
AuthName "Who are you?"
AuthType basic
AuthUserFile /var/www/html/helmreleaseAPItest/.htpasswd
Require valid-user

# ll /etc/httpd/conf.d/helm.conf
-rw-r--r--. 1 root root 157 Dec 21 11:18 /etc/httpd/conf.d/helm.conf
# cat /etc/httpd/conf.d/helm.conf
Alias /helm/ "/var/www/html/helmreleaseAPItest/"

<Directory /var/www/html/helmreleaseAPItest>
  AllowOverride AuthConfig
  Require all granted
</Directory>

<Files "helmreleaseAPItest/gbapp-0.1.0.tgz">
  AuthName "Who are you?"
  AuthType basic
  AuthUserFile /var/www/html/helmreleaseAPItest/.htpasswd
  Require valid-user
</Files>

5. Restart httpd
# systemctl restart httpd

File: default-e2e-test-data/testcases/release-operator-e2e.json [hunk 1]
```diff
@@ -4,5 +4,12 @@
   "desc": "helmrelease install test: guestbook deploy",
   "urls": ["https://raw.githubusercontent.com/open-cluster-management/multicloud-operators-subscription-release/e2e-test-setup/examples/test-guestbook010.yaml"],
   "target_cluster": "hub"
+},
+
+{
+  "test_id": "release-003",
+  "desc": "helmrelease with secret install test: guestbook deploy",
+  "url": ["https://raw.githubusercontent.com/pezhang/helm-release-API-test/gh-pages/test-guestbook010-secret.yaml"],
```
Comment (ianzhang366) [inline default-e2e-test-data/testcases/release-operator-e2e.json:12]: over here, it's `urls` instead of `url`, the key for your URL

Comment (pezhang) [reply default-e2e-test-data/testcases/release-operator-e2e.json:12]: @ianzhang366 I have fixed it. Thanks.

**Comment by ianzhang366** [comment]
Hi @pezhang 

Thanks for the PR, everything looks great!

However, in a long term, we might need to find a private chart that host by ACM
team for this test... otherwise, you or us will have to maintain the Http
server for this test, what do you say?

**Comment by pezhang** [comment]
@ianzhang366  I agree with you. It's better we deploy the chart to a fixed ACM server. And set the basic auto for the Apache server as above.

Does ACM team have such a server? Maybe I can help deploy the Apache and chart on this server or VM on this server.

**Comment by ianzhang366** [comment]
@pezhang 

while checking another issue, I saw this,
https://github.com/open-cluster-management/app-ui-e2e-private-helm/tree/master/mychart

basically, they used a private github repo as a private chart. I guess, we can
do the same, what be do you think? If you like the idea, I can help you set up
the private chart. Then you can change your PR a bit... 

Let me know what do you think

**Comment by pezhang** [comment]
Due to patch conflicts, and I would prefer to open a new PR to re-send this test case. Close this one.

**Pull request closed without merging**